src/kinematics.cpp

```diff
@@ -0,0 +1,174 @@
+#include <reuleaux/kinematics.h>
+
+#if IK_VERSION > 54
+#define IKREAL_TYPE IkReal // for IKFast 56,61
+#else
+#define IKREAL_TYPE IKReal // for IKFast 54
+#endif
+
+
+
+
+using namespace std;
+IKREAL_TYPE eerot[9],eetrans[3];
+namespace kinematics{
+
+
+float Kinematics::SIGN(float x) {
+    return (x >= 0.0f) ? +1.0f : -1.0f;
+}
+
+float Kinematics::NORM(float a, float b, float c, float d) {
+    return sqrt(a * a + b * b + c * c + d * d);
+}
+
+
+void Kinematics::getPoseFromFK(const std::vector<double> joint_values,  std::vector<double>& pose){
+
+     #if IK_VERSION > 54
+    // for IKFast 56,61
+    unsigned int num_of_joints = GetNumJoints();
+    unsigned int num_free_parameters = GetNumFreeParameters();
+#else
+    // for IKFast 54
+    unsigned int num_of_joints = getNumJoints();
+    unsigned int num_free_parameters = getNumFreeParameters();
+#endif
+    IKREAL_TYPE joints[num_of_joints];
+    
+    //cout<<joint_values[2]<<endl;
+    
+    for (unsigned int i=0; i<num_of_joints; i++)
+        {
+	   
+            joints[i] = joint_values[i];
+            
+        }
+#if IK_VERSION > 54
+    // for IKFast 56,61
+        ComputeFk(joints, eetrans, eerot); // void return
+#else
+        // for IKFast 54
+        fk(joints, eetrans, eerot); // void return
+#endif
+     //cout<<"translation: "<<eetrans[0]<<" "<<eetrans[1]<<" "<<eetrans[2]<<endl;
+	// Convert rotation matrix to quaternion (Daisuke Miyazaki)
+        float q0 = ( eerot[0] + eerot[4] + eerot[8] + 1.0f) / 4.0f;
```

Comment:
There is a [kdl_conversions](https://github.com/ros/geometry/tree/indigo-devel/kdl_conversions) and a [tf_conversions](https://github.com/ros/geometry/tree/indigo-devel/tf_conversions) package for performing some of these calculations.  Use those and only create your own for converting between custom messages/types.